### PR TITLE
Security Settings: Fix for unsupported plans

### PR DESCRIPTION
**Pull request by allilevine**
#### Changes proposed in this Pull Request

* Use `security-settings` feature to determine whether to show the security settings page

#### Testing instructions

* Go to Settings > General for a free, personal, or premium Atomic site

* Without this change, you can access the Security page via the URL (/settings/security/SITE_URL). With this change, you should get an error page:

Before | After
------ | -----
Screen Shot 2021-06-07 at 5 12 20 PM | Screen Shot 2021-06-07 at 5 11 30 PM

* Regression: you should still see the page on Jetpack sites and Atomic sites with supported plans. Simple sites should behave the same as free Atomic.
* Note that there will still be a Settings > Security nav item, that will be hidden by this Jetpack PR: https://github.com/Automattic/jetpack/pull/20384

Related to #51820

### CODE REVIEW

**Comment by pablinos** [comment]
A few random thoughts on this:

- Is it just free plans that we want to hide the settings from? It should probably only be available to sites that have the ability to install plugins. Although it can't happen at the moment, we will likely have Atomic sites with any type of plan in the future, so we might want to account for that now. Maybe this should be another feature like the 'set primary domain' one? @retrofox might have opinions there 😄 
- Should we ensure that the security settings are set to the default when the plan changes? This is probably a separate change, but if a security option is disabled, as the plan downgrades, we should probably enable it to give the site the same managed protection as simple sites
- This shuts off the settings in the UI, but could they be changed using the API still? If so, we should probably disable that as a follow-up PR too.

**Comment by allilevine** [comment]
> * Is it just free plans that we want to hide the settings from? It should probably only be available to sites that have the ability to install plugins. Although it can't happen at the moment, we will likely have Atomic sites with any type of plan in the future, so we might want to account for that now. Maybe this should be another feature like the 'set primary domain' one? @retrofox might have opinions there 😄

Good point! I'm a little confused though about whether sites on Personal and Premium will have this issue where they return `true` for Jetpack features they shouldn't have. I guess it depends if we're checking the plan or just `is_jetpack`.

> * Should we ensure that the security settings are set to the default when the plan changes? This is probably a separate change, but if a security option is disabled, as the plan downgrades, we should probably enable it to give the site the same managed protection as simple sites
> 
> * This shuts off the settings in the UI, but could they be changed using the API still? If so, we should probably disable that as a follow-up PR too.

Thanks, definitely necessary follow-ups!

**Comment by pablinos** [comment]
> I guess it depends if we're checking the plan or just is_jetpack.

Yes, that's right. I think in this instance, that is what's happening. It's another one of those situations where the plan and the features available aren't being checked, as it hasn't been necessary previously.

**Comment by allilevine** [comment]
Noting that there's no longer a Settings nav bar as of https://github.com/Automattic/wp-calypso/pull/53824
Security Settings are now in a submenu: https://github.com/Automattic/wp-calypso/pull/53824

File: client/my-sites/site-settings/settings-security/test/main.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/**
```
Comment (allilevine) [inline client/my-sites/site-settings/settings-security/test/main.jsx:1]: These throw a warning because `requestSupportHistory` is missing an `onError` handler:
https://github.com/Automattic/wp-calypso/blob/e314a251e948361d9019441d2976041f51fed054/client/state/data-layer/wpcom/help/support-history/index.js#L27-L30
But it seems like we have a lot of tests that do, and I'd prefer not to change unrelated code in this PR.

**Comment by retrofox** [comment]
should we wait until https://github.com/Automattic/wp-calypso/pull/53196 lands? They seem to introduce similar changes, for instance, adding the query component to the layout.

File: client/my-sites/site-settings/settings-security/test/main.jsx (hunk 1 above)
Comment (pablinos) [reply client/my-sites/site-settings/settings-security/test/main.jsx:1]: Agreed. Let's create an issue to sort out the error handling.

**Review by pablinos — COMMENTED** [review]
I'm not sure if this should be a follow-up change or not, but I have a test site that doesn't have site title, and that leaves the button saying 'Manage general settings for'. Should we use the site URL in this situation like we do in the sidebar?

image

File: client/my-sites/site-settings/settings-security/test/main.jsx (hunk 1 above)
Comment (allilevine) [reply client/my-sites/site-settings/settings-security/test/main.jsx:1]: Thanks, opened an issue here: https://github.com/Automattic/wp-calypso/issues/54663

**Comment by allilevine** [comment]
> I'm not sure if this should be a follow-up change or not, but I have a test site that doesn't have site title, and that leaves the button saying 'Manage general settings for'. Should we use the site URL in this situation like we do in the sidebar?

Sounds good, I added the fix here since it seems like a small change: https://github.com/Automattic/wp-calypso/pull/53484/commits/464de06d38a1121e8791cb60b94f0b4f4d3f8d99

**Comment by retrofox** [comment]
> > I'm not sure if this should be a follow-up change or not, but I have a test site that doesn't have site title, and that leaves the button saying 'Manage general settings for'. Should we use the site URL in this situation like we do in the sidebar?
> 
> Sounds good, I added the fix here since it seems like a small change: [464de06](https://github.com/Automattic/wp-calypso/commit/464de06d38a1121e8791cb60b94f0b4f4d3f8d99)

it seems we are going to show the site slug in the UI. Maybe not ideal but better than nothing :-)

**Review by retrofox — APPROVED** [review]
LGTM. Tested in Staging and Live branch, and it works as expected. I like the idea of replacing the code with the features stuff. Thanks for this fix.

staging (now) | live branch
-----|-------
image | image

**Comment by pablinos** [comment]
> it seems we are going to show the site slug in the UI. Maybe not ideal but better than nothing :-)

Yeah it's better than nothing and we're already doing that in the sidebar. We could change the button text to simply 'Manage general settings' and remove the 'for ....' when there's no title?

**Comment by retrofox** [comment]
> > it seems we are going to show the site slug in the UI. Maybe not ideal but better than nothing :-)
> 
> Yeah it's better than nothing and we're already doing that in the sidebar. We could change the button text to simply 'Manage general settings' and remove the 'for ....' when there's no title?

That sounds good to me, or maybe transforming the slug, in snake/kebab case, to UpperCamelCase (aka PascalCase ;-))
